shotover/src/transforms/query_counter.rs

```diff
@@ -28,7 +28,9 @@ impl QueryCounter {
     pub fn new(counter_name: String) -> Self {
         counter!("shotover_query_count", "name" => counter_name.clone());
 
-        QueryCounter { counter_name }
+        QueryCounter {
+            counter_name: counter_name.leak(),
```

Comment:
Lets add a comment to reassure readers that leaking is fine in this case. Something like:
```rust
// Leaking here is fine since the builder is created only once during shotover startup.
```